### ISSUE TITLE
gcc warns of "Function defined but not used"

### ISSUE THREAD

**Issue by cinnaboot** [badaix/aixlog#5]
Hello, thanks for your header only logger. I've found it very nice so far.

I do like to compile with -Wall when using gcc, and I get these warnings apparently from using internally linked functions.

Would you consider adding something like: https://gist.github.com/cinnaboot/80cc12ffa6095c8db325be631fc1bb1f to supress the warnings?

I got the macro from this thread https://cboard.cprogramming.com/cplusplus-programming/118695-help-remove-warning-x-function-defined-but-not-used.html that also suggests inlining to get around it.

**Comment by badaix** [comment]
will these macros work on all platforms?
Alternatively, one could use all of them once:

```diff
diff --git a/include/aixlog.hpp b/include/aixlog.hpp
index 35bf73e..8fc6f8e 100644
--- a/include/aixlog.hpp
+++ b/include/aixlog.hpp
@@ -442,6 +442,8 @@ static std::ostream& operator<< (std::ostream& os, const Timestamp& timestamp);
 static std::ostream& operator<< (std::ostream& os, const Tag& tag);
 static std::ostream& operator<< (std::ostream& os, const Function& function);
 static std::ostream& operator<< (std::ostream& os, const Conditional& conditional);
+static std::ostream& operator<< (std::ostream& os, const Color& color);
+static std::ostream& operator<< (std::ostream& os, const TextColor& text_color);
 
 using log_sink_ptr = std::shared_ptr<Sink>;
 
@@ -470,8 +472,6 @@ public:
 
                for (const auto& sink: log_sinks)
                        Log::instance().add_logsink(sink);
-
-               std::clog.rdbuf(&Log::instance());
        }
 
        template<typename T, typename... Ts>
@@ -536,6 +536,8 @@ public:
 protected:
        Log() noexcept
        {
+               std::clog.rdbuf(this);
+               std::clog << Severity() << Type::normal << Tag() << Function() << Conditional() << AixLog::Color::NONE << std::flush;
        }
 
        int sync() override
```

**Comment by cinnaboot** [comment]
I assume you would have to keep adding to the macro for each compiler that throws warnings.

I can at least confirm that your patch works for gcc 7.3.0, and is probably less of a hack :P

**Comment by badaix** [comment]
fixed with [v1.1.0](https://github.com/badaix/aixlog/tree/v1.1.0)
